src/utils/mem_component.c

```diff
@@ -80,6 +107,31 @@ xccl_status_t xccl_mem_component_reduce(void *sbuf1, void *sbuf2, void *target,
     return mem_components[mem_type]->reduce(sbuf1, sbuf2, target, count, dtype, op);
 }
 
+xccl_status_t
+xccl_mem_component_reduce_multi(void *sbuf1, void *sbuf2, void *rbuf, size_t count,
+                                size_t size, size_t stride, xccl_dt_t dtype,
+                                xccl_op_t op, ucs_memory_type_t mem_type)
+{
+    int i;
+
+    assert(count > 1);
```

Comment:
Why this assertion? count = 1 should be the case of radix=2 - you should have seen this assertion failure, no?
Imho, count = 1 should just work as is.

Reply:
count=1 will work, I just thought that there are no reasons to use multireduce in this case, but actually radix=2 is the case

Reply:
And we didn't see assert because it was release build. With --enable-debug and radix=2 it checks and fails